webpack/assets/javascripts/react_app/components/SettingsTable/SettingsTableHelpers.js

```diff
@@ -111,19 +111,26 @@ const formatArraySelection = (attr, setting) => {
     return null;
   }
 
-  const [, model] = setting[attr].split('-');
-
-  const selectGroup = selectValues.find(group => group.groupLabel === model);
-
-  if (!selectGroup) {
-    return null;
-  }
-
-  const selectedItem = selectGroup.children.find(
-    item => item.value === setting[attr]
-  );
-
-  return selectedItem && selectedItem.label;
+  const findSelectValue = (attrValue, valuesSelection) => {
+    const foundItem = valuesSelection.reduce((memo, group) => {
+      if (memo) {
+        return memo;
+      }
+
+      if (group.value === attrValue) {
+        return group;
+      }
+
+      if (group.children) {
+        return group.children.find(child => child.value === attrValue);
+      }
+      return null;
+    }, null);
+
+    return foundItem ? foundItem.label : null;
+  };
+
+  return findSelectValue(setting[attr], selectValues);
```

Comment:
How about something like this which avoids an internal function and returns once we found our value:
```suggestion
  for (let group of selectValues){
    if (group.value === setting[attr]) {
      return group?.label;
    }
    if (group.children) {
      return group.children.find(child => child.value === setting[attr])?.label;
    }

  }

  return null;
```